content/en/blog/_posts/2021-XX-XX-writing-a-controller-for-pod-labels.md

```diff
@@ -0,0 +1,473 @@
+            pod.Labels = make(map[string]string)
+        }
+        pod.Labels[podNameLabel] = pod.Name
+        log.Info("adding label")
+    } else {
+        // If the label should not be set but is, remove it.
+        delete(pod.Labels, podNameLabel)
+        log.Info("removing label")
+    }
+```
+
+Finally, let's push our updated Pod to the Kubernetes API:
+
+```go
+    /*
+        Step 2: Update the Pod in the Kubernetes API.
+    */
+
+    if err := r.Update(ctx, &pod); err != nil {
+        log.Error(err, "unable to update Pod")
+        return ctrl.Result{}, err
+    }
+```
+
+When writing our updated Pod to the Kubernetes API, there is a risk that the Pod
+has been updated or deleted since we first read it. When writing a Kubernetes
+controller, we should keep in mind that we are not the only actors in the
+cluster. When this happens, the best thing to do is start the reconciliation
+from scratch, by requeuing the event. Lets do exactly that:
+
+```go
+    /*
+        Step 2: Update the Pod in the Kubernetes API.
+    */
+
+    if err := r.Update(ctx, &pod); err != nil {
+        if apierrors.IsConflict(err) {
+            // The Pod has been updated since we read it.
+            // Requeue the Pod to try to reconciliate again.
+            return ctrl.Result{Requeue: true}, nil
+        }
+        if apierrors.IsNotFound(err) {
+            // The Pod has been deleted since we read it.
+            // Requeue the Pod to try to reconciliate again.
+            return ctrl.Result{Requeue: true}, nil
+        }
+        log.Error(err, "unable to update Pod")
+        return ctrl.Result{}, err
+    }
+```
+
+Let's remember to return successfully at the end of the method:
+
+```go
+    return ctrl.Result{}, nil
+}
+```
+
+And that's it! We are now ready to run the controller on our cluster.
+
+## Run the controller on your cluster
+
+To run our controller on your cluster, we need to run the operator. For that,
+all you will need is `kubectl`. If you don't have a Kubernetes cluster at hand,
+I recommend you start one locally with [KinD (Kubernetes in Docker)][kind].
+
+All it takes to run the operator from your machine is this command:
+
+```bash
+make run
+```
+
+After a few seconds, you should see the operator's logs. Notice that our
+controller's `Reconcile` method was called for all Pods already running in the
+cluster.
+
+Let's keep the operator running and, in another terminal, create a new Pod:
+
+```bash
+kubectl run --image=nginx my-nginx
+```
+
+The operator should quickly print some logs, indicating that it reacted to the
+Pod's creation and subsequent changes in status:
+
+```text
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+INFO    controllers.Pod no update required  {"pod": "default/my-nginx"}
+```
+
+Lets check the Pod's labels:
```

Comment:
Where it means "let us", replace /Lets/Let's/ and s/lets/let's/. (multiple instances)

Reply:
```suggestion
Check the pod's labels:
```

Reply:
@busser, you can ignore this suggestion.